python/ray/serve/tests/test_batching.py

```diff
@@ -214,6 +215,39 @@ async def call(arg):
         await t3
 
 
+@pytest.mark.asyncio
+async def test_batch_timeout_empty_queue():
+    """Check that Serve waits when creating batches.
+
+    Serve should wait a full batch_wait_timeout_s after receiving the first
+    request in the next batch before processing the batch.
+    """
+
+    @serve.batch(max_batch_size=10, batch_wait_timeout_s=0.1)
+    async def no_op(requests):
+        return ["No-op"] * len(requests)
+
+    num_iterations = 2
+    for iteration in range(num_iterations):
+        tasks = [get_or_create_event_loop().create_task(no_op(None)) for _ in range(9)]
+        done, _ = await asyncio.wait(tasks, timeout=0.01)
+
+        # Due to the long timeout, none of the tasks should finish until a tenth
+        # request is submitted
+        assert len(done) == 0
+
+        tasks.append(get_or_create_event_loop().create_task(no_op(None)))
+        done, _ = await asyncio.wait(tasks, timeout=0.01)
+
+        # All the timeout tasks should be finished
+        assert set(tasks) == set(done)
+        assert all(t.result() == "No-op" for t in tasks)
+
+        if iteration < num_iterations - 1:
+            # Leave queue empty for batch_wait_timeout_s between batches
```

Comment:
oh I get it now -- without the change, this would immediately return the first request enqueued

Reply:
Yeah, exactly.